applications/visualizer/frontend/src/components/viewers/TwoD/TwoDMenu.tsx

```diff
@@ -0,0 +1,246 @@
+// GraphMenu.tsx
+import {useState} from 'react';
+import {
+    IconButton,
+    Popover,
+    Typography,
+    Box,
+    ToggleButtonGroup,
+    ToggleButton, Switch, FormControlLabel, FormGroup, FormControl, FormLabel, Tooltip, TextField
+} from '@mui/material';
+import ZoomInIcon from '@mui/icons-material/ZoomIn';
+import ZoomOutIcon from '@mui/icons-material/ZoomOut';
+import SettingsIcon from '@mui/icons-material/Settings';
+import VisibilityIcon from '@mui/icons-material/Visibility';
+import DownloadIcon from '@mui/icons-material/Download';
+import RestartAltIcon from '@mui/icons-material/RestartAlt';
+import RemoveIcon from '@mui/icons-material/Remove';
+import AddIcon from '@mui/icons-material/Add';
+import {GRAPH_LAYOUTS, ZOOM_DELTA} from "../../../settings/twoDSettings.tsx";
+import {applyLayout} from "../../../helpers/twoD/twoDHelpers.ts";
+import {ColoringOptions} from "../../../helpers/twoD/coloringStrategy/ColoringStrategy.ts";
+
+const TwoDMenu = ({
+                      cyRef,
+                      layout,
+                      onLayoutChange,
+                      coloringOption,
+                      onColoringOptionChange,
+                      includeNeighboringCells,
+                      setIncludeNeighboringCells,
+                      includeNeighboringCellsAsIndividualCells,
+                      setIncludeNeighboringCellsAsIndividualCells,
+                      includeAnnotations,
+                      setIncludeAnnotations,
+                      thresholdChemical,
+                      setThresholdChemical,
+                      thresholdElectrical,
+                      setThresholdElectrical
+                  }) => {
+    const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+
+    const onZoomIn = () => {
+        if (cyRef.current) {
```

Comment:
Is it possible to have only `cyRef` instead of `cyRef.current`? It feels strange to receive as parameter of a component a kind of low-level implementation object. This would make sense if the relation `cyRef.current` would be changed in the component (and it would be also weird to have a side-effect in the component).